### ISSUE TITLE
Hide modifiers are all set to enabled on new game when using a profile

### ISSUE THREAD

**Issue by camsta4** [dguzek/Simply-Love-SM5#15]
1. Set a local profile
2. Start a new game
3. Play at least 1 song, fail or pass
4. Exit the Select Music screen, passing via Summary, Ranking and Game Over screens
5. Start a new game

The Hide row in the Extra Modifiers screen has all options set ENABLED, regardless of what they were set to prior to finishing the previous game.

Confirmed on both Windows 8.1 with SM 5.0.6 and Windows XP with SM 5.0.5.

**Comment by dguzek** [comment]
Thank you for pointing this out.  At this time, I'm fairly perplexed by what I am observing happen...

This is the function that handles the "Hide" OptionRow:
```
function OptionRowHide()
	local t = {
		Name = "Hide",
		LayoutType = "ShowAllInRow",
		SelectType = "SelectMultiple",
		OneChoiceForAllPlayers = false,
		ExportOnChange = false,
		Choices = { "Targets", "Background", "Combo", "Life", "Score" },
		LoadSelections = function(self, list, pn)
			local mods = SL[ToEnumShortString(pn)].ActiveModifiers
			list[1] = mods.HideTargets or false
			list[2] = mods.HideSongBG or false
			list[3] = mods.HideCombo or false
			list[4] = mods.HideLifebar or false
			list[5] = mods.HideScore or false
			
			SM("player: " .. pn)
			SM("list at the end of LoadSelections(): ")
			SM(list)
			SM("-----------------")
		end,
		SaveSelections = function(self, list, pn)
			SM("player: " .. pn)
			SM("list at the beginning of SaveSelections(): ")
			SM(list)
			SM("-----------------")
			
			local mods = SL[ToEnumShortString(pn)].ActiveModifiers
			mods.HideTargets = list[1]
			mods.HideSongBG = list[2]
			mods.HideCombo = list[3]
			mods.HideLifebar = list[4]
			mods.HideScore = list[5]
			ApplyHide(pn)
		end,
	}
	return t
end
```

Noting that *SM()* is a recursive logging convenience function, here is the out put I get:
```
03:47.318: player: PlayerNumber_P1
03:47.318: list at the end of LoadSelections(): 
03:47.318: key: 1, val: false
03:47.318: key: 2, val: true
03:47.318: key: 3, val: false
03:47.318: key: 4, val: true
03:47.318: key: 5, val: false
03:47.318: 
03:47.318: -----------------
03:47.339: player: PlayerNumber_P1
03:47.339: list at the beginning of SaveSelections(): 
03:47.339: key: 1, val: true
03:47.339: key: 2, val: true
03:47.339: key: 3, val: true
03:47.339: key: 4, val: true
03:47.339: key: 5, val: true
03:47.339: 
03:47.339: -----------------
```
In the above log, I am using a profile for PLAYER_1 to load *true* values for *HideSongBG* and *HideLifebar*.  The logs generated from the LoadSelections() function check out okay.

I had thought that the state of the *list* table was supposed to be carried from LoadSelections() into SaveSelections().  Normally, this seems to happen, but the logs above and the behavior shown in StepMania clearly demonstrate that this does not happen here.  All the list values are suddenly *true*.  I'm not sure having a profile loaded has this impact.

Maybe my understanding of how this works is incorrect.  Perhaps @kyzentun can chime in here...?

**Comment by kyzentun** [comment]
It's important to include type information in your logging functions, because the type of a value matters.  In this case, your modifiers are empty strings for some unknown reason, and the empty string is equivalent to true.
Here's the output I got when I put in logging with my rec_print_table:
```
01:22.190: PushScreen: "ScreenPlayerOptions2"
01:22.197: mods in LoadSelections:
01:22.197: (string)SpeedMod: (string)800
01:22.197: (string)ScreenFilter: (string)Off
01:22.197: (string)Mini: (string)Normal
01:22.197: (string)Vocalization: (string)Shizuka (Japanese)
01:22.197: (string)HideCombo: (string)
01:22.197: (string)HideScore: (string)
01:22.197: (string)HideLifebar: (string)
01:22.197: (string)JudgmentGraphic: (string)Love
01:22.197: (string)HideSongBG: (string)
01:22.197: (string)HideTargets: (string)
01:22.197: (string)SpeedModType: (string)M
01:22.197: end
01:22.197: list in LoadSelections:
01:22.197: (number)1: (boolean)false
01:22.197: (number)2: (boolean)false
01:22.197: (number)3: (boolean)false
01:22.197: (number)4: (boolean)false
01:22.197: (number)5: (boolean)false
01:22.197: end
01:22.197: list after LoadSelections:
01:22.197: (number)1: (string)
01:22.197: (number)2: (string)
01:22.197: (number)3: (string)
01:22.197: (number)4: (string)
01:22.197: (number)5: (string)
01:22.197: end
01:22.205: mods in SaveSelections:
01:22.205: (string)SpeedMod: (string)800
01:22.205: (string)ScreenFilter: (string)Off
01:22.205: (string)Mini: (string)Normal
01:22.205: (string)Vocalization: (string)Shizuka (Japanese)
01:22.205: (string)HideCombo: (string)
01:22.205: (string)HideScore: (string)
01:22.205: (string)HideLifebar: (string)
01:22.205: (string)JudgmentGraphic: (string)Love
01:22.205: (string)HideSongBG: (string)
01:22.205: (string)HideTargets: (string)
01:22.205: (string)SpeedModType: (string)M
01:22.205: end
01:22.205: list in SaveSelections:
01:22.205: (number)1: (boolean)true
01:22.205: (number)2: (boolean)true
01:22.205: (number)3: (boolean)true
01:22.205: (number)4: (boolean)true
01:22.205: (number)5: (boolean)true
01:22.205: end
01:22.205: mods after SaveSelections:
01:22.205: (string)SpeedMod: (string)800
01:22.205: (string)ScreenFilter: (string)Off
01:22.205: (string)Mini: (string)Normal
01:22.205: (string)Vocalization: (string)Shizuka (Japanese)
01:22.205: (string)HideCombo: (boolean)true
01:22.205: (string)HideScore: (boolean)true
01:22.205: (string)HideLifebar: (boolean)true
01:22.205: (string)JudgmentGraphic: (string)Love
01:22.205: (string)HideSongBG: (boolean)true
01:22.205: (string)HideTargets: (boolean)true
01:22.205: (string)SpeedModType: (string)M
01:22.205: end
```

**Comment by dguzek** [comment]
It's looking like anything read from a .cfg file via ./Scripts/SL-CustomProfiles.lua is a string.  I'll look into that soon.

Thanks!

**Comment by kyzentun** [comment]
You know, there's a reason why my settings system exists....

**Comment by dguzek** [comment]
:+1:

**Comment by dguzek** [comment]
This issue should be resolved as of 301de86b388828d3fd521a56aeb7c46693cac870
(As I note in the commit message, there are cleaner solutions, but it works for now.)

Please try it out, @camsta4 , and reply here.

**Comment by camsta4** [comment]
Issue's fixed, thanks!
